[Llama 3.1] Updates MLLOG tags

Elnifio opened the pull request.

ShriyaPalsamudram commented:
@Elnifio can you verify that the logs from this reference indeed pass the latest checkers?

mmarcinkiewicz commented on large_language_model_pretraining/nemo/callbacks.py:
```diff
@@ -124,11 +124,11 @@ def log_metrics(self, metrics, step):
 
     def log_validation_loss(self, metrics, step):
-        consumed_tokens = (step - self.init_global_step) * self.gbs * self.seq_len
+        consumed_tokens = step * self.gbs
```
shouldn't that be called `consumed_samples`?

Elnifio replied:
Updated in the latest commit. Thanks for catching that!

mmarcinkiewicz commented on large_language_model_pretraining/nemo/callbacks.py:
```diff
@@ -202,12 +202,12 @@ def on_train_end(self, trainer, pl_module):
     def on_validation_start(self, trainer, pl_module):
-        mllogger.end(key=constants.BLOCK_STOP, metadata={'epoch_num': self.consumed_tokens(trainer)})
-        mllogger.start(key=constants.EVAL_START, metadata={'epoch_num': self.consumed_tokens(trainer)})
+        mllogger.end(key=constants.BLOCK_STOP, metadata={constants.SAMPLES_COUNT: self.consumed_tokens(trainer)})
```
not sure where `self.consumed_tokens()` is defined, but does it return tokens or samples? If tokens, we need to switch it to samples. If samples, we need to rename it

Elnifio commented:
@ShriyaPalsamudram @mmarcinkiewicz the latest run fails the RCP checker in the following test:  
 - [common.yaml line 128](https://github.com/mlcommons/logging/blob/16136821819c64aed584a6be8174aaeeaea7e2e5/mlperf_logging/compliance_checker/training_5.0.0/common.yaml#L128), missing `train_samples` - this is **fixed** in my latest commit, where I renamed `trained_samples` to `train_samples`
 - [closed_llama31_405b.yaml line 35](https://github.com/mlcommons/logging/blob/16136821819c64aed584a6be8174aaeeaea7e2e5/mlperf_logging/compliance_checker/training_5.0.0/closed_llama31_405b.yaml#L35), incorrect `opt_learning_rate_decay_schedule` value - this is **fixed** in my latest commit, where I renamed `cosine with linear warmups` to `cosine with linear warmup`. 
 - [common.yaml line 97, 103](https://github.com/mlcommons/logging/blob/16136821819c64aed584a6be8174aaeeaea7e2e5/mlperf_logging/compliance_checker/training_5.0.0/common.yaml#L97): missing `epoch_num` in epoch_start, epoch_end tags - I'm wondering whether, similar to other tags, we should accept both `epoch_num` and `samples_count` in the metadata. What do you think?

Elnifio commented:
> @ShriyaPalsamudram @mmarcinkiewicz the latest run fails the RCP checker in the following test:
> 
> * [common.yaml line 128](https://github.com/mlcommons/logging/blob/16136821819c64aed584a6be8174aaeeaea7e2e5/mlperf_logging/compliance_checker/training_5.0.0/common.yaml#L128), missing `train_samples` - this is **fixed** in my latest commit, where I renamed `trained_samples` to `train_samples`
> * [closed_llama31_405b.yaml line 35](https://github.com/mlcommons/logging/blob/16136821819c64aed584a6be8174aaeeaea7e2e5/mlperf_logging/compliance_checker/training_5.0.0/closed_llama31_405b.yaml#L35), incorrect `opt_learning_rate_decay_schedule` value - this is **fixed** in my latest commit, where I renamed `cosine with linear warmups` to `cosine with linear warmup`.
> * [common.yaml line 97, 103](https://github.com/mlcommons/logging/blob/16136821819c64aed584a6be8174aaeeaea7e2e5/mlperf_logging/compliance_checker/training_5.0.0/common.yaml#L97): missing `epoch_num` in epoch_start, epoch_end tags - I'm wondering whether, similar to other tags, we should accept both `epoch_num` and `samples_count` in the metadata. What do you think?

Further updating on this comment - I have ran the RCP checker against the latest compliance checker from [PR 414](https://github.com/mlcommons/logging/pull/414/files) and it passed without any errors / warnings, so I'd say both PRs are ready to merge.
